pkg/registry/registry.go

```diff
@@ -0,0 +1,211 @@
+package registry
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/suse-edge/edge-image-builder/pkg/fileio"
+	"github.com/suse-edge/edge-image-builder/pkg/http"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+	"go.uber.org/zap"
+)
+
+type helmClient interface {
+	AddRepo(repository *image.HelmRepository) error
+	RegistryLogin(repository *image.HelmRepository) error
+	Pull(chart string, repository *image.HelmRepository, version, destDir string) (string, error)
+	Template(chart, repository, version, valuesFilePath, kubeVersion, targetNamespace string) ([]map[string]any, error)
+}
+
+type helmChart struct {
+	image.HelmChart
+	localPath     string
+	repositoryURL string
+}
+
+type Registry struct {
+	embeddedImages []image.ContainerImage
+	manifestsDir   string
+	helmClient     helmClient
+	helmCharts     []*helmChart
+	helmValuesDir  string
+	kubeVersion    string
+}
+
+func New(ctx *image.Context, localManifestsDir string, helmClient helmClient, helmValuesDir string) (*Registry, error) {
+	manifestsDir, err := storeManifests(ctx, localManifestsDir)
+	if err != nil {
+		return nil, fmt.Errorf("storing manifests: %w", err)
+	}
+
+	charts, err := storeHelmCharts(ctx, helmClient)
+	if err != nil {
+		return nil, fmt.Errorf("storing helm charts: %w", err)
+	}
+
+	return &Registry{
+		embeddedImages: ctx.ImageDefinition.EmbeddedArtifactRegistry.ContainerImages,
+		manifestsDir:   manifestsDir,
+		helmClient:     helmClient,
+		helmCharts:     charts,
+		helmValuesDir:  helmValuesDir,
+		kubeVersion:    ctx.ImageDefinition.Kubernetes.Version,
+	}, nil
+}
+
+func (r *Registry) ManifestsPath() string {
+	return r.manifestsDir
+}
+
+func storeManifests(ctx *image.Context, localManifestsDir string) (string, error) {
+	var manifestsPathPopulated bool
+	manifestsDestDir := filepath.Join(ctx.BuildDir, "manifests")
```

Comment:
Personal Preference: I imagine this directory name is only used in this instance, hence it's not a constant. I prefer the consistency of having all of the directory structure names as constants. 

To be clear, I have zero programmatic reason for liking this, it just feels clean to me :)